ros_ethercat_model/include/ros_ethercat_model/robot_state.hpp

```diff
@@ -85,12 +86,15 @@ class RobotState : public hardware_interface::HardwareInterface
            it != robot_model_.joints_.end();
            ++it)
       {
-	if (use_joint_(it->second->name) && (it->second->type == urdf::Joint::PRISMATIC || it->second->type == urdf::Joint::REVOLUTE))
+	if (use_joint_(it->second->name) && (it->second->type == urdf::Joint::PRISMATIC ||
+                                             it->second->type == urdf::Joint::REVOLUTE))
 	{
           joint_states_[it->first].joint_ = it->second;
 	}
       }
 
+      imu_states_["rh_imu"] = ImuState( "rh_imu", "rh_palm");
```

Comment:
Can we scan the joint name list to look for existing prefixes?
As it is, It wouldn't work with a left hand.

Reply:
yeah, guess we could